### PR TITLE
Adding Alert V2 improvements.

### PR DESCRIPTION
**Pull request by puneetjaiswal**
Separated Data and Anomaly Reports modules

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java [hunk 1]
```diff
@@ -184,7 +184,9 @@ public Response generateAndSendAlertForDatasets(@PathParam("startTime") Long sta
     configuration.setSmtpConfiguration(smtpConfiguration);
     configuration.setDashboardHost(teHost);
 
-    anomalyReportGenerator.buildReport(startTime, endTime, anomalies, configuration, includeSentAnomaliesOnly, toAddr, fromAddr);
+    anomalyReportGenerator
+        .buildReport(startTime, endTime, anomalies, "Thirdeye Anomaly Report", configuration,
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java]: Add these Strings to constants?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/alert/util/DataReportHelper.java [hunk 2]
```diff
@@ -21,27 +23,37 @@
...
   // report list metric vs groupByKey vs dimensionVal vs timeBucket vs values
-  public static List<MetricDimensionReport> getDimensionReportList(
-      List<ContributorViewResponse> reports) {
+  public List<MetricDimensionReport> getDimensionReportList(List<ContributorViewResponse> reports) {
     List<MetricDimensionReport> ultimateResult = new ArrayList<>();
     for (ContributorViewResponse report : reports) {
       MetricDimensionReport metricDimensionReport = new MetricDimensionReport();
       String metric = report.getMetrics().get(0);
+      List<MetricConfigDTO> metricConfigs = metricDAO.findByMetricName(metric);
+      String dataset = "Unknown";
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/alert/util/DataReportHelper.java]: Why defaulting to "Unknown"? is it for display purposes?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/alert/util/AlertFilterHelper.java [hunk 3]
```diff
@@ -18,5 +27,66 @@ public static AlertFilter getAlertFilter(AlertFilterType filterType) {
         return new DummyAlertFilter();
     }
   }
+
+  /**
+   *
+   * @param results
+   * @return
+   */
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/alert/util/AlertFilterHelper.java]: Can you finish the JavaDoc?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java (hunk 1 above)
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/EmailResource.java]: this string is not used at more than one place, perf. wise no extra cost.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/alert/util/DataReportHelper.java (hunk 2 above)
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/alert/util/DataReportHelper.java]: Yes. Ideally we would never use this default.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/alert/util/AlertFilterHelper.java (hunk 3 above)
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/alert/util/AlertFilterHelper.java]: sure.

**Pull request merged**